File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        ans = float("inf")
        left = 0
        right = len(nums) - 1
        while left <= right:
            m = (left + right) // 2
            if nums[m] < ans:
                ans = nums[m]
            if nums[left] < nums[right]:
                right = m - 1
            elif nums[right] < nums[left]:
                if nums[left] > nums[m]:
                    right = m - 1
                else:
                    left = m + 1
            else:
                return ans
        return ans

File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {
        if (nums.length == 1){
            return nums[0];
        }
        Map<Integer, Integer> check = new HashMap<>();
        int ans = 0;
        for(int num : nums){
            if (check.get(num) == null) {
                check.put(num, 1);
            }
            else {
                check.put(num, check.get(num) + 1);
                if(check.get(num) > (nums.length / 2)) ans = num;
            }
        }    
        System.out.println(check);
        return ans;
    }
}